### PR TITLE
Make functional test configuration more flexible

### LINKED ISSUE #717: kubevirt/cirros-registry-disk-demo:devel is missing
When trying to run the functional tests I'm getting the following error

```
Failed to pull image "kubevirt/cirros-registry-disk-demo:devel": rpc error: code = 2 desc = Tag devel not found in repository docker.io/kubevirt/cirros-registry-disk-demo
````

Looking at `https://hub.docker.com/r/kubevirt/cirros-registry-disk-demo/tags/`, the image is indeed missing.

### PR DESCRIPTION
**Pull request by rmohr**
Add two flags to the test.tests binary:

 * -tag (defaults to "latest")
 * -prefix (defaults to "kubevirt")

"-tag" allows testing specific docer versions,  "-prefix" allow pulling
images from different repos (e.g. "-prefix localhost:5000/kubevirt").

Fixes #717. @gbenhaim FYI

Signed-off-by: Roman Mohr <rmohr@redhat.com>

### CODE REVIEW

File: tests/utils.go [hunk 1]
```diff
@@ -807,3 +809,22 @@ func NewConsoleExpecter(virtCli kubecli.KubevirtClient, vm *v1.VirtualMachine, c
 		Check: func() bool { return true },
 	}, timeout, opts...)
 }
+
+type RegistryDisk string
+
+const (
+	CirrosRegistryDisk RegistryDisk = "cirros"
```
Comment (cynepco3hahue) [inline tests/utils.go]: Can you change all constants to form
RegistryDiskCirros
RegistryDiskAlpine
it just more easy to view all registry disk with auto-completion

Comment (rmohr) [reply tests/utils.go]: Done. :+1:

File: tests/utils.go [hunk 2]
```diff
@@ -88,21 +98,14 @@ const (
 )
 
 const (
-	labelISCSIPod         = "iscsi-demo-target"
-	labelISCSIWithAuthPod = "iscsi-auth-demo-target"
-	labelNFSPod           = "nfs-server-demo"
+	labelNFSPod = "nfs-server-demo"
```
Comment (davidvossel) [inline tests/utils.go]: This NFS pod doesn't exist anymore. Whatever references it should be able to be removed safely at this point. It's just dead code.

File: tests/utils.go [hunk 3]
```diff
@@ -88,21 +98,14 @@ const (
...
 )
 
 const (
 	iscsiIqn        = "iqn.2017-01.io.kubevirt:sn.42"
```
Comment (davidvossel) [inline tests/utils.go]: oops, i need to remove this too.

File: hack/dockerized [hunk 4]
```diff
@@ -44,7 +44,7 @@ _rsync() {
 rm ${KUBEVIRT_DIR}/.glide*.hash -f
 
 # Copy kubevirt into the persistent docker volume
-_rsync --delete --exclude 'cluster/vagrant*' --exclude "_out" --exclude ".vagrant" ${KUBEVIRT_DIR}/ "rsync://root@127.0.0.1:${RSYNCD_PORT}/build"
+_rsync --delete --exclude 'cluster/**/.kubectl' --exclude 'cluster/**/.oc' --exclude 'cluster/**/.kubeconfig' --exclude "_out" --exclude ".vagrant" ${KUBEVIRT_DIR}/ "rsync://root@127.0.0.1:${RSYNCD_PORT}/build"
```
Comment (davidvossel) [inline hack/dockerized]: what's the context here for why some files in cluster should be synced.

Comment (rmohr) [reply hack/dockerized]: The reason is pretty dump: The build step also formats all code, including bash scripts. If we don't pass in all bash scripts in there they don't get formatted.

**Comment by gbenhaim** [comment]
@rmohr +1, thanks!

File: hack/dockerized (hunk 4 above)
Comment (cynepco3hahue) [reply hack/dockerized]: I hope we will merge it as soon as possible, it just too annoying to checkout bash scripts after each `make` command.

**Pull request merged**